### PR TITLE
[SPARK-47933][PYTHON] Parent Column class for Spark Connect and Spark Classic

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

Same as https://github.com/apache/spark/pull/46129 but for `Column` class.

### Why are the changes needed?

Same as https://github.com/apache/spark/pull/46129

### Does this PR introduce _any_ user-facing change?

Same as https://github.com/apache/spark/pull/46129

### How was this patch tested?

Manually tested, and CI should verify them.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Review by zhengruifeng — APPROVED** [review]

**Review by dongjoon-hyun — COMMENTED** [review]
What is the meaning of `pyspark.sql.classic`?

**Comment by HyukjinKwon** [comment]
Let me run this naming stuff separately (maybe sending an email to the dev mailing list) because Spark Classic has been used in other places lately in any event.

**Comment by dongjoon-hyun** [comment]
Sure, please proceed it. We can discuss the naming separately, @HyukjinKwon .

File: python/pyspark/ml/stat.py [hunk 1]
```diff
@@ -22,7 +22,8 @@
 from pyspark.ml.common import _java2py, _py2java
 from pyspark.ml.linalg import Matrix, Vector
 from pyspark.ml.wrapper import JavaWrapper, _jvm
-from pyspark.sql.column import Column, _to_seq
+from pyspark.sql.column import Column
```
[the remark this answers is not in the document]
Comment (HyukjinKwon) [reply python/pyspark/ml/stat.py:25]: This file is only used in Spark Classic so it should probably be fine. It is a bit messy and complicated .. I will try to refactor them separately.

File: python/pyspark/ml/connect/functions.py [hunk 2]
```diff
@@ -15,7 +15,7 @@
 # limitations under the License.
 #
 from pyspark.ml import functions as PyMLFunctions
-from pyspark.sql.connect.column import Column
+from pyspark.sql.column import Column
```
[the remark this answers is not in the document]
Comment (HyukjinKwon) [reply python/pyspark/ml/connect/functions.py:18]: It is actually related to the type hints. We're using `pyspark.sql.Column` for type hints, and when we actually need to create the instance, we should use `pyspark.sql.classic.column import Column` or `pyspark.sql.connect.column import Column` .. it is also a bit messy. I should probably clean those up separately.

**Comment by HyukjinKwon** [comment]
Merged to master.

For naming, I will send an email soon.
For others, please leave the comments. I will followup.

**Pull request closed without merging**